src/gsy_e/models/strategy/external_strategies/__init__.py

```diff
@@ -316,6 +320,117 @@ def _get_market_from_time_slot(self, time_slot: DateTime) -> MarketBase:
+                "transaction_id": arguments.get("transaction_id"),
+                "area_uuid": self.device.uuid,
+                "message": response_message}
+        except (AssertionError, GSyException):
+            logging.exception("Failed to post PV offer.")
+            response = {
+                "command": "offer", "status": "error",
+                "error_message": "Error when handling offer create "
+                                 f"on area {self.device.name} with arguments {arguments}.",
+                "area_uuid": self.device.uuid,
+                "transaction_id": arguments.get("transaction_id")}
+        return response
+
+    def _bid_aggregator_impl(
+            self, arguments: Dict, market: "MarketBase", time_slot: DateTime,
+            required_energy: float) -> Dict:
+        """Post bid in the market for aggregator connection to load or PV."""
+        response_message = ""
+        arguments, filtered_fields = self.filter_degrees_of_freedom_arguments(arguments)
+        if filtered_fields:
+            response_message = (
+                "The following arguments are not supported for this market and have been "
+                f"removed from your order: {filtered_fields}.")
+
+        required_args = {"price", "energy", "type", "transaction_id"}
+        allowed_args = required_args.union({"replace_existing",
+                                            "time_slot",
+                                            "attributes",
+                                            "requirements"})
+
+        try:
+            # Check that all required arguments have been provided
+            assert all(arg in arguments.keys() for arg in required_args)
+            # Check that every provided argument is allowed
+            assert all(arg in allowed_args for arg in arguments.keys())
+
+            replace_existing = arguments.pop("replace_existing", True)
+            assert self.can_bid_be_posted(
+                arguments["energy"],
+                arguments["price"],
+                required_energy,
+                market,
+                time_slot=time_slot,
+                replace_existing=replace_existing)
+            bid = self.post_bid(
+                market,
+                arguments["price"],
+                arguments["energy"],
+                replace_existing=replace_existing,
+                attributes=arguments.get("attributes"),
+                requirements=arguments.get("requirements")
+            )
+            response = {
+                "command": "bid", "status": "ready",
+                "bid": bid.to_json_string(replace_existing=replace_existing),
+                "area_uuid": self.device.uuid,
+                "transaction_id": arguments.get("transaction_id"),
+                "message": response_message}
+        except (AssertionError, GSyException):
+            logging.exception("Error when handling bid on area %s", self.device.name)
```

Comment:
Even better than my class name proposal on the previous comment, you could copy the same error message on the logging.exception in the corresponding offer exception log.